test/versioned/redis/redis-v4.tap.js

```diff
@@ -208,12 +213,13 @@ test('Redis instrumentation', function (t) {
       t.ok(agent.getTransaction(), 'should not lose transaction state')
       transaction.end()
       verify()
+      t.end()
     })
 
     function verify() {
       const setSegment1 = transaction.trace.root.children[0]
-      const selectSegment = transaction.trace.root.children[2]
-      const setSegment2 = transaction.trace.root.children[4]
+      const selectSegment = transaction.trace.root.children[1]
+      const setSegment2 = transaction.trace.root.children[2]
```

Comment:
I don't see an explanation for this change. Did changing the order in which end is called reorder some transactions?

Reply:
It's in the PR details 

> I also refactored the tap tests to explicity call t.end(). There was a mix of t.plan, t.autoend and t.end. It seemed to cause inconsistencies in test runs. When I changed that the segment array in the last test changed because of the setTimeouts tap creates when calling t.autoend.

Reply:
Oh, the "segment array", of course.